geo/src/algorithm/cross_track_distance.rs

```diff
@@ -0,0 +1,128 @@
+use num_traits::FromPrimitive;
+use geo_types::{CoordFloat, Point};
+use crate::{Bearing, HaversineDistance, MEAN_EARTH_RADIUS};
+
+/// Determine the cross track distance (also known as the cross track error) which is the shortest
+/// distance between a point and a continuous line.
+pub trait CrossTrackDistance<T, Rhs = Self> {
+    /// Determine the cross track distance between this point and a line
+    /// which passes through line_point_a and line_point_b
+    ///
+    /// # Units
+    ///
+    /// - return value: meters
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use geo::prelude::*;
+    /// use geo::point;
+    ///
+    /// // New York City
+    /// let p1 = point!(x: -74.006f64, y: 40.7128f64);
+    ///
+    /// // Miami
+    /// let line_point_a = point!(x: -80.1918f64, y: 25.7617f64);
+    ///
+    /// // Washington
+    /// let line_point_b = point!(x: -120.7401, y: 47.7511f64);
+    ///
+    /// let distance = p1.cross_track_distance(&line_point_a, &line_point_b);
+    ///
+    /// assert_eq!(
+    ///     1_547_104., // meters
+    ///     distance.round()
+    /// );
+    /// ```
+    fn cross_track_distance(&self, line_point_a: &Rhs, line_point_b: &Rhs) -> T;
+}
+
+impl<T> CrossTrackDistance<T, Point<T>> for Point<T>
+    where
+        T: CoordFloat + FromPrimitive,
+{
+    fn cross_track_distance(&self, line_point_a: &Point<T>, line_point_b: &Point<T>) -> T {
+        let mean_earth_radius = T::from(MEAN_EARTH_RADIUS).unwrap();
+        let l_delta_13: T = line_point_a.haversine_distance(self) / mean_earth_radius;
+        let theta_13: T = line_point_a.bearing(*self).to_radians();
+        let theta_12: T = line_point_a.bearing(*line_point_b).to_radians();
+        let l_delta_xt: T = (l_delta_13.sin() * (theta_12 - theta_13).sin()).asin();
+        mean_earth_radius * l_delta_xt.abs()
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::CrossTrackDistance;
+    use crate::HaversineDistance;
+    use crate::Point;
+
+    #[test]
+    fn should_equal_haversine_distance_for_short_lines() {
```

Comment:
Can you explain what this test is about? 

The current test name seems misleading - the fact that they are equal is not because the line is short, but because the point p is "approximately" perpendicular to the line, so that the haversine distance is "almost" the same as the cross track distance.

Said differently, this test would also pass if we made the line very long:

```
let line_point_a = Point::new(1., 100.);
let line_point_b = Point::new(1., -100.);
```

And conversely, a very short line, depending on where it's placed, could fail this test:
```
let line_point_a = Point::new(10., 10.);
let line_point_b = Point::new(9.99, 9.99);
```

Is that right, or have I misunderstood something? 

If so it seems redundant with  `cross_track_distance_to_line_orthogonal_to_point` below. I'd say delete it or at least give it a clearer name like `cross_track_distance_to_line_orthogonal_to_point_with_short_line` or whatever.

Reply:
Hi @michaelkirk you're absolutely right, this is effectively a pointless variation of `cross_track_distance_to_line_orthogonal_to_point`, I think it's best to remove it.